src/main/java/com/linkedin/xinfra/monitor/services/ProduceService.java

```diff
@@ -87,7 +87,11 @@ public ProduceService(Map<String, Object> props, String name) throws Exception {
     _partitioner = config.getConfiguredInstance(ProduceServiceConfig.PARTITIONER_CLASS_CONFIG, KMPartitioner.class);
     _threadsNum = config.getInt(ProduceServiceConfig.PRODUCE_THREAD_NUM_CONFIG);
     _topic = config.getString(ProduceServiceConfig.TOPIC_CONFIG);
-    _producerId = config.getString(ProduceServiceConfig.PRODUCER_ID_CONFIG);
+    // TODO: Some producer implementation does not allow multiple producers with the same producer ID. So, this is a
+    // temporary way to avoid producer ID collision. A better way to fix this issue is to let users provide producer ID
+    // prefix and KMF keeps track of the the producer service creation process and dynamically generate producer ID using
+    // the given prefix.
+    _producerId = String.format("%s-%s", _name, config.getString(ProduceServiceConfig.PRODUCER_ID_CONFIG));
```

Comment:
should we honor the producer id set by config? 
this will confuse users if users intended to use a specific producer id and it was changed internally by the Produce service

Reply:
This change was intended to fix an existing issue. I will remove this change from this PR since it's not related to other changes in this PR. The producer ID issue will need another PR to fix. Thanks.